.github/workflows/feature-build-hniesb.yml

```diff
@@ -1,36 +1,154 @@
-# This is a basic workflow to help you get started with Actions
+# This is a manual workflow that performs following activities:
+# 1. Determine the version number tag i.e. FeatureBuild-<branch-name>-<build execution>  
+# 2. Determine tags for OpenShift image
+# 3. Run the maven install for common code from input branch name
+# 4. Run the maven install for hnsesb code from input branch name
+# 5. Run the docker file to create image
+# 6. Push the image to OpenShift repository
+# 7. Deploy the image in Dev environment
 
-name: hnsesb-feature-build
+
+name: feature-build-hnsesb
+
+# Build the image and deploy in OpenShift test environment
+env:
+  OPENSHIFT_SERVER: ${{ secrets.C5839F_OPENSHIFT_SERVER }}
+  OPENSHIFT_TOKEN: ${{ secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_TOKEN }}
+  DEFAULT_NAMESPACE: c5839f-tools
+  DEV_NAMESPACE: c5839f-dev
+  IMAGE_REGISTRY: image-registry.apps.silver.devops.gov.bc.ca
+  IMAGE_REGISTRY_USER: ${{ secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_USER }}
+  IMAGE_NAME: hnsesb
+  IMAGE_TAGS: 
+  FEATUREBUILD_NAME:
 
 # Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
-
-  # Allows you to run this workflow manually from the Actions tab
+  # The user specifies the optional tag  
   workflow_dispatch:
+    inputs:
+      BRANCH_NAME:
+        description: 'Name of the feature branch'
+        required: true
+        default: main
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  # This job runs the maven commands to compile code and build package.
+  # This job also creates docker image and push the image to OpenShift repository
+  createHniEsbFeatureBuild:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      OPENSHIFT_IMAGE_SHA: ${{ steps.push-image.outputs.digest }}
+      FEATUREBUILD_NAME: ${{ env.FEATUREBUILD_NAME }}
```

Comment:
Would FEATURE_BUILD_NAME make more sense in terms of readability (where _ equals space).

Reply:
updated